node/pkg/watchers/solana/client.go

```diff
@@ -741,7 +808,7 @@ func (s *SolanaWatcher) processInstruction(ctx context.Context, logger *zap.Logg
 		return false, fmt.Errorf("failed to determine commitment: %w", err)
 	}
 
-	if level != s.commitment {
+	if level != s.commitment && !isReobservation {
```

Comment:
remind me of the rationale for this change? maybe worth a comment here of what this is doing

Reply:
This is being addressed with PR#4231.